### PR TITLE
SLCORE-740 Telemetry for sharing connected mode

### PR DESCRIPTION
**Pull request by nquinquenel**

### CODE REVIEW

File: backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/ShareConnectedModePayload.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import com.google.gson.annotations.SerializedName;
+import javax.annotation.Nullable;
+
+public class ShareConnectedModePayload {
+
+  @SerializedName("manual_added_bindings_count")
+  public final Integer manualAddedBindingsCount;
+
+  @SerializedName("imported_added_bindings_count")
```
Comment (sophio-japharidze-sonarsource) [inline backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/ShareConnectedModePayload.java:30]: minor note - I think `imported_bindings_count` & `auto_bindings_count` would be enough and also in line with what DPO [expects](https://docs.google.com/spreadsheets/d/1oIPd5dZrjWqtujEQ5hQIJ_fVe3heCUIzmNmEaoAC_W4/edit#gid=368366020)

**Review by sophio-japharidze-sonarsource — APPROVED** [review]
The changes look good to me. The only question is if all these fields should indeed be nested under `shared_connected_mode` payload? And if there is any preference for the exact field keys. I think both are questions for our DPO.

File: backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/ShareConnectedModePayload.java (hunk 1 above)
Comment (nquinquenel) [reply backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/ShareConnectedModePayload.java:30]: DPO needs the total binding count, which to get we need to add the manual binding counts. I don't think we can skip it, WDYT?

Comment (sophio-japharidze-sonarsource) [reply backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/ShareConnectedModePayload.java:30]: Yes yes for sure, I just meant for the naming `imported_added_bindings_count` -> `imported_bindings_count`

Comment (nquinquenel) [reply backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/payload/ShareConnectedModePayload.java:30]: oh I see 😄

**Review by vojtech-suchy-sonarsource — APPROVED** [review]
LGTM.

As discussed, I find it a bit odd that we stopped cleaning the telemetry data after the telemetry ping. Thank you for confirming it with DPOs. 

Also, a reminder that we will need to update the telemetry samples + [[Data Team](https://xtranet-sonarsource.atlassian.net/wiki/spaces/DATA?homepageId=2237989083)
[Data sources, catalogs and lineage](https://xtranet-sonarsource.atlassian.net/wiki/spaces/DATA/pages/2348187685/Data+sources+catalogs+and+lineage)
[Telemetry data catalog for SonarLint](https://xtranet-sonarsource.atlassian.net/wiki/spaces/DATA/pages/2347991072/Telemetry+data+catalog+for+SonarLint#) (unless DPO will do it)





Summarize



Share

Telemetry data catalog](https://xtranet-sonarsource.atlassian.net/wiki/spaces/DATA/pages/2347991072/Telemetry+data+catalog+for+SonarLint)

**Pull request merged**